src/convertToObject.js

```diff
@@ -6,7 +6,36 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const splitThis = sourceString.split(';');
+  const result = {};
+
+  for (const oneString of splitThis) {
+    let word = '';
+    let wordTwo = '';
+    let fazeTwo = false;
+
+    for (const symbol of oneString) {
+      if (symbol.toLowerCase() !== symbol.toUpperCase() || symbol === '-') {
+        if (fazeTwo !== true) {
+          word += symbol;
+        }
+      }
+
+      if (fazeTwo === true) {
+        wordTwo += symbol;
+      }
+
+      if (symbol === ':') {
+        fazeTwo = true;
+      }
+    }
+
+    if (word.length !== 0) {
+      result[word] = wordTwo.trim();
+    }
+  }
+
+  return result;
```

Comment:
The function uses a `for...of` loop to iterate over the string and build the object. This approach is not wrong, but it does not follow the checklist item 5, which suggests avoiding loops and using iteration methods instead. Consider using `split`, `map`, and `reduce` or `forEach` to transform the string into an object in a more functional style.